### ISSUE TITLE
completion and error check donesn't work

### ISSUE THREAD

**Issue by saintaugustinus** [jorgenschaefer/elpy#436]
I am new to python and elpy
I found a very strange problem that when i first open a *.py file the elpy start but the completion and error check don't work. 
But when __i close the file and reopen it (or manually restart python-mode again)__ the completion and error check work fine.

This is the minibuffer shows:
````
Completion needs an inferior Python process running
````
and here is my configure:
````
(defun ext/elpy-setup ()
  (setq elpy-rpc-python-command "python2")
  (setq elpy-rpc-backend "jedi")
  (elpy-use-ipython)
  (when (require 'flycheck nil t)
    (remove-hook 'elpy-modules 'elpy-module-flymake)))

(add-hook 'elpy-mode-hook 'ext/elpy-setup)
(add-hook 'python-mode-hook 'elpy-mode)

````

and my emacs version is 24.4

**Comment by jorgenschaefer** [comment]
Hello, and thanks for the report.

The error in the minibuffer is from default python.el, not from Elpy – can you make sure that Elpy is actually enabled when you open the file for the first time?

**Comment by saintaugustinus** [comment]
Yes i am sure it is enabled when i open the file for the first time
I found that __pyvenv-mode__ do not start at the first time  and it is not closed after i close the file
This mode is enabled after i reopen the file or manually restart python-mode

**Comment by jorgenschaefer** [comment]
How did you check that Elpy is enabled?

**Comment by saintaugustinus** [comment]
the elpy is show an the status bar 
and 
````
M-x elpy-mode 
````
can turn off it
and i notice that if i restart elpy-mode the things work fine

**Comment by jorgenschaefer** [comment]
This is really weird. Your symptoms are all as if Elpy is not enabled at all, but apparently it is enabled. I have no idea what's going on there.

What is the output of `M-x elpy-config` for you on a freshly started Emacs?

(Just as a note, this is unlikely to be related: Changing `elpy-modules` that late in the initialization process will not have the desired effect. You should do that, and probably most of the rest of your configuration, after elpy.el is loaded.)

**Comment by saintaugustinus** [comment]
I have tried to change my config and start emacs with the '-q' option and just load elpy's config unfortunately it's still not fix my problem.
here is my new config:
````
(setq elpy-rpc-python-command "python2") ;;Python2 is my choice
(eval-after-load 'elpy
  '(progn
     (setq elpy-rpc-backend "jedi")
     (setq python-indent-offset 6)
     (elpy-use-ipython)
     (when (require 'flycheck nil t)
       (remove-hook 'elpy-modules 'elpy-module-flymake))))

(add-hook 'python-mode-hook 'elpy-mode)
````

here is the output of ````M-x elpy-config ````:
````
Virtualenv........: None
RPC Python........: 2.7.9 (/usr/bin/python2)
Interactive Python: ipython (/usr/bin/ipython)
Emacs.............: 24.4.1
Elpy..............: 1.6.0
Jedi..............: 0.8.1-final0
Rope..............: Not found (0.10.2 available)
Syntax checker....: pyflakes (/usr/bin/pyflakes)
````
and the first time i open python file:
````
Virtualenv........: None
RPC Python........: 2.7.9 (/usr/bin/python2)
Interactive Python: ipython (/usr/bin/ipython)
Emacs.............: 24.4.1
Elpy..............: 1.6.0
Jedi..............: 0.8.1-final0
Rope..............: Not found (0.10.2 available)
Syntax checker....: pyflakes (/usr/bin/pyflakes)
````
and after elpy works:
````
Virtualenv........: None
RPC Python........: 2.7.9 (/usr/bin/python2)
Interactive Python: ipython (/usr/bin/ipython)
Emacs.............: 24.4.1
Elpy..............: 1.6.0
Jedi..............: 0.8.1-final0
Rope..............: Not found (0.10.2 available)
Syntax checker....: pyflakes (/usr/bin/pyflakes)
````
it is totally the same

**Comment by saintaugustinus** [comment]
Hey I think I found where the problem is 
I manually remove the elpy-module-highlight-indentation from elpy-modules everything works fine now!
There is still a problem that the pyvenv-mode doesn't close while elpy-mode is close, so i have to manually disable it.
Is there anyway to make it automatically enable and disable with elpy.

**Comment by jorgenschaefer** [comment]
Interesting effect – there is a bug in the unstable release of highlight-indentation-mode from MELPA (see #435), are you using that?

As for pyvenv-mode, I'm afraid that is a global mode, so it does not make sense to disable it when a single Elpy buffer is closed. It is possible to disable it when the last Elpy buffer closes, but then you'd lose the virtualenv activation for file-local variables. Out of curiosity, why do you want to disable it?

**Comment by saintaugustinus** [comment]
Yes I using melpa and i got the same debug infomation.
And the melpa version is still not fix  #394 which you have fixed (I use flycheck instead of flymake)
As for pyvenv-mode I just thought it is a bug at the beginning

**Comment by saintaugustinus** [comment]
The problem fixed after new highlight-indentation-mode's release
